Automattic/wp-calypso
# Migrate 2FA push notification polling away from Redux and data layer

This PR rewrites the `PushNotificationApprovalPoller` component that sends a repeated request to `/wp-login.php` until it gets a "success" response, meaning that the user approved the login in their mobile app.

Instead of using Redux and data layer, the component is now completely self-contained, doing a local `fetch` loop.

The loop is abortable using the browser's `AbortController` API. However, after some consideration, I decided to _not_ pass the `signal` object to the `fetch` call, making it abortable itself. That's mainly because there's also the `remoteLoginUser` call, which is more difficult to abort. So, if I can't abort everything reliably, I decided to check the "aborted" condition only at one place in the loop.

This rewrite allowed me to remove a lot of Redux and data layer code, and this functionality was also the last one to use the `HTTP_REQUEST` action, i.e., data layer requests that do direct `fetch` instead of going through WP.com REST Proxy.

**How to test:**
Try to log in on an account with enabled 2FA and active mobile app (I guess that virtually every a11n account satisfies that). Verify that the `wp-login.php` requests are really sent in a loop and that after approval the loop finishes.

I had trouble to verify the "successful login" case though: when testing on `calypso.localhost`, my mobile app has never shown the notification.

Verify also that the loop is really aborted when you navigate away from the poll screen.
